core/sync/index.js

```diff
@@ -676,6 +641,32 @@ class Sync {
     return p
   }
 
+  // Wait for a change in PouchDB's changesfeed after the given sequence and
+  // with the expected synced path.
+  //
+  // We should be careful to not hold a lock while we wait for this change and
+  // that it will indeed be merged at some point or we will end up waiting
+  // forever.
+  async waitForNewChangeOn(seq /*: number */, expectedPath /*: string */) {
+    return new Promise((resolve, reject) => {
+      const opts = {
+        live: true,
+        limit: 1,
+        since: seq,
+        filter: '_view',
+        view: 'byPath',
+        return_docs: false,
+        include_docs: true
+      }
+      this.pouch.db
+        .changes(opts)
+        .on('change', ({ doc }) => {
+          if (doc.path === expectedPath) resolve()
```

Comment:
I think we should add `else reject()` on this line. We call the changes feed with `limit: 1`, so if this change doesn't match, we won't have other changes, and the promise will never be resolved/rejected without the `else` branch.

Reply:
It's a `live` subscription to the changes feed so every change will be notified. The `limit: 1` option here makes sure we only receive on change at a time.